go/types/struct.go

```diff
@@ -195,13 +194,87 @@ func EscapeStructField(input string) string {
 		buf.WriteString(hs)
 		return buf.String()
 	}
+	return escapeFields(input, encode)
+}
+
+func CamelCaseFieldName(input string) string {
+	encode := func(s1 string, p *regexp.Regexp) string {
+		if p.MatchString(s1) {
+			return s1
+		}
+		return ""
+	}
 
+	splitFields := strings.Fields(input)
+	output := escapeFields(splitFields[0], encode)
+
+	if len(splitFields) > 1 {
+		output = strings.ToLower(output)
+	}
+
+	for _, field := range splitFields[1:] {
+		output += strings.Title(strings.ToLower(escapeFields(field, encode)))
+	}
+
+	if !IsValidStructFieldName(output) {
```

Comment:
Can you put a little comment here explaining how this can happen? The code is correct, but it's not immediately obvious from reading it why this would occur.